File: linealATramos.py
#!/bin/python3
# Ian Mu;oz Nu;ez - Funcion 'Lineal a Tramos'

import numpy as np
import matplotlib.pyplot as plt

n = 1000
v = np.linspace(-10, 10, n) # Variable independiente

phi = np.zeros(n) # Funcion o variable dependiente
for i in range(n):
    if v[i] >= 1:
        phi[i] = 1
    elif v[i] > 0 and v[i] < 1:
        phi[i] = v[i]
    else:
        phi[i] = 0

plt.figure(1)
plt.grid()

plt.plot(v, phi, 'g-', linewidth=2)

plt.title("Funcion lineal a tramos", fontsize=20)
plt.xlabel('v', fontsize=15)
plt.ylabel('$\phi(v)$', fontsize=15)

plt.show()

File: linealATramos.m
% Ian Mu;oz Nu;ez - Funcion 'Lineal a Tramos'

close all
clear
clc

v = linspace(-10, 10, 1000); % Variable independiente

y = []; % Funcion o variable dependiente
dy = []; % Derivada de la funcion
for i= 1:1000
    if v(i) >= 1
        y(i) = 1;
        dy(i) = 0;
    elseif v(i) > 0 && v(i) < 1
        y(i) = v(i);
        dy(i) = 1;
    else
        y(i) = 0;
        dy(i) = 0;
    end
end

figure(1)

subplot(2,1,1)
hold on
grid on

plot(v, y, 'g-', 'LineWidth', 2)
title("Funcion lineal a tramos", 'FontSize', 20)
xlabel('v', 'FontSize', 15)
ylabel('\phi(v)', 'FontSize', 15)

subplot(2,1,2)
hold on
grid on

plot(v, dy, 'r-', 'LineWidth', 2)
title("Derivada de la funcion", 'FontSize', 20)
xlabel('v', 'FontSize', 15)
ylabel('d \phi(v)', 'FontSize', 15)

